src/openms/source/SYSTEM/File.cpp

```diff
@@ -65,6 +65,31 @@ using namespace std;
 namespace OpenMS
 {
 
+  File::TempDir::TempDir(bool keep_dir)
```

Comment:
now the nitpicking starts, sorry in advance :)

You could make use of the Ctor's initializer list and remove some redundant lines, which would make this a bit more readable (IMHO!)

```
File::TempDir::TempDir(bool keep_dir)
  : keep_dir_(keep_dir)
{
    temp_dir_ = File::getTempDirectory() + "/" + File::getUniqueName() + "/";

    OPENMS_LOG_DEBUG << "Creating temporary directory '" << temp_dir_ << "'" << std::endl;
    QDir().mkpath(temp_dir_.toQString());
  };
```

the `QDir::toNativeSeparators` is technically correct, but usually does not provide any benefit (Windows can deal fine with / in paths). I vaguely remember there was a disadvantage doing \, but I cannot recall right now...
But thats a matter of style.